### PR TITLE
igl | vulkan | fix transform error between igl::TextureFormat and VkFormat about compressed texture

### PR DESCRIPTION
**Pull request by vinsentli**

### CODE REVIEW

File: src/igl/vulkan/util/TextureFormat.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
  */
 
 #include <igl/vulkan/util/TextureFormat.h>
+#include <igl/vulkan/Common.h>
```
Comment (corporateshark) [inline src/igl/vulkan/util/TextureFormat.cpp:9]: The purpose of `util/TextureFormat.cpp` is to be used where Vulkan headers are not compiled in. That's why we need this macro definitions.

**Review by corporateshark — CHANGES_REQUESTED** [review]

File: src/igl/vulkan/util/TextureFormat.cpp (hunk 1 above)
Comment (vinsentli) [reply src/igl/vulkan/util/TextureFormat.cpp]: done

File: src/igl/vulkan/Common.cpp [hunk 2]
```diff
@@ -156,69 +156,69 @@ VkFormat textureFormatToVkFormat(igl::TextureFormat format) {
...
+    return VK_FORMAT_ASTC_12x12_UNORM_BLOCK;
   case TextureFormat::SRGB8_A8_ASTC_12x12:
-    return VK_FORMAT_UNDEFINED;
+    return VK_FORMAT_ASTC_12x12_SRGB_BLOCK;
   case TextureFormat::RGBA_PVRTC_2BPPV1:
-    return VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG;
+    return VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG;
   case TextureFormat::RGB_PVRTC_2BPPV1:
-    return VK_FORMAT_UNDEFINED;
+    return VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG;
```
Comment (corporateshark) [inline src/igl/vulkan/Common.cpp:217]: Shouldn't it be `VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG`?

File: src/igl/vulkan/Common.cpp [hunk 3]
```diff
@@ -156,69 +156,69 @@ VkFormat textureFormatToVkFormat(igl::TextureFormat format) {
...
   case TextureFormat::RGBA_PVRTC_4BPPV1:
-    return VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG;
+    return VK_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG;
   case TextureFormat::RGB_PVRTC_4BPPV1:
-    return VK_FORMAT_UNDEFINED;
+    return VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG;
```
Comment (corporateshark) [inline src/igl/vulkan/Common.cpp:221]: `UNORM`?

**Review by corporateshark — CHANGES_REQUESTED** [review]

File: src/igl/vulkan/Common.cpp (hunk 2 above)
Comment (vinsentli) [reply src/igl/vulkan/Common.cpp]: Yes, you are right.

https://github.com/KhronosGroup/Vulkan-Docs/issues/512

IMG_texture_compression_pvrtc

GL_COMPRESSED_RGB_PVRTC_4BPPV1_IMG -> VK_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGB_PVRTC_2BPPV1_IMG -> VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG -> VK_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_2BPPV1_IMG -> VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG
IMG_texture_compression_pvrtc2

GL_COMPRESSED_RGBA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_UNORM_BLOCK_IMG
EXT_pvrtc_sRGB

GL_COMPRESSED_SRGB_PVRTC_2BPPV1_EXT -> VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_PVRTC_4BPPV1_EXT -> VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV1_EXT -> VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV1_EXT -> VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_SRGB_BLOCK_IMG

File: src/igl/vulkan/Common.cpp (hunk 3 above)
Comment (vinsentli) [reply src/igl/vulkan/Common.cpp]: Yes, UNORM

**Comment by vinsentli** [comment]
https://github.com/KhronosGroup/Vulkan-Docs/issues/512

IMG_texture_compression_pvrtc

GL_COMPRESSED_RGB_PVRTC_4BPPV1_IMG -> VK_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGB_PVRTC_2BPPV1_IMG -> VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG -> VK_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_2BPPV1_IMG -> VK_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG
IMG_texture_compression_pvrtc2

GL_COMPRESSED_RGBA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_UNORM_BLOCK_IMG
GL_COMPRESSED_RGBA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_UNORM_BLOCK_IMG
EXT_pvrtc_sRGB

GL_COMPRESSED_SRGB_PVRTC_2BPPV1_EXT -> VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_PVRTC_4BPPV1_EXT -> VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV1_EXT -> VK_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV1_EXT -> VK_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV2_IMG -> VK_FORMAT_PVRTC2_2BPP_SRGB_BLOCK_IMG

GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV2_IMG -> VK_FORMAT_PVRTC2_4BPP_SRGB_BLOCK_IMG

**Pull request closed without merging**